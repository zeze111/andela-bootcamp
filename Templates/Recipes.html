<!DOCTYPE html>
<html lang="en">

<head>
  <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1.0" />
  <title>More-Recipes</title>

  <!-- CSS  -->
  <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
  <link href="assets/css/materialize.css" type="text/css" rel="stylesheet" media="screen,projection" />
  <link href="assets/css/style.css" type="text/css" rel="stylesheet" media="screen,projection" />


</head>

<body class="lime lighten-5">
  <nav class="teal lighten-2" role="navigation">
    <div class="nav-wrapper"> <a id="logo" href="#" class="left brand-logo">More-Recipes</a>
      <ul id="nav-mobile" class="side-nav">
        <li><a href="#">Navbar Link</a></li>
      </ul>
      <a href="#" data-activates="nav-mobile" class="button-collapse"><i class="material-icons">menu</i></a>
      <ul class="right hide-on-med-and-down">
        <li><a class="btn-floating btn-medium waves-effect waves-light grey">
            <i class="material-icons">add</i></a></li>
        <li><a class="btn-floating btn-medium waves-effect waves-light grey">
                <i class="material-icons">star_border</i></a></li>
        <li><a href="#">Sign In</a></li>
      </ul>
    </div>
  </nav>

  <div class="container z-depth-4">
    <div class="row" style="margin-top:50px;">
      <div class="col s2 offset-s1">
        <div class="card" style="width:230px; height:230px;">
          <div class="card-image">
            <img src="assets/images/card.jpeg" class="materialboxed responsive-img">
          </div>
        </div>
      </div>
      <div class="col s8" style="margin-left:30px;">
        <div class="hreview-aggregate">
          <div class="row" style="margin-bottom:0px; margin-top:70px;">
            <div class="col s12 push-s1">
              <h5 id="title" style="margin-bottom:0px; margin-top:0px;"> Baked Alaska </h5>
            </div>
            <div class="col s12 push-s1">
              <p style=" margin-top:0px;"> Posted by <a id="uname" href="#"> Username </a> </p>
            </div>
            <div class="col s12 push-s1">
              <p id="desc" style=" margin-top:0px;">Recipe description </p>
            </div>
            <div class="col s10 push-s1">
              <p style=" margin-bottom:0px;"> Like this Recipe? Add to your favourites
                <a class="btn-floating btn-small waves-effect waves-light yellow">
                     <i class="material-icons">star_border</i></a>
              </p>
            </div>
            <div class="col s5 push-s1">
              <a href="#rev" style="margin-top:0px;" class="section scrollspy"> Reviews </a>
            </div>
          </div>
        </div>
      </div>

      <div class="row">
        <div class="col s2 pull-s2">
          <a class="btn-floating waves-effect waves-light blue">
                <i class="material-icons">thumb_up</i></a>
          <p id="likes" class="col s1 push-s5">3</p>
        </div>
        <div class="col s2 pull-s2">
          <a class="btn-floating waves-effect waves-light blue">
                  <i class="material-icons">thumb_down</i></a>
          <p id="likes" class="col s1 push-s5">1</p>
        </div>
      </div>
    </div>

    <div class="container">
      <h5 class="left-align" style="margin-left:50px"> Recipe </h5>
      <div id="time" class="col s6 ">
        <p style="font-size:18px; margin-bottom:0px;"> Prep Time: </p>
        <p id="preptime" style="margin-top:0px;"> Time </p>
      </div>
      <div id="Ing" class="col s6 ">
        <p style="font-size:18px; margin-left:50px"> Ingredients: </p>
        <p id="ingred"> Ingredients info </p>
      </div>
      <div id="Ins" class="col s6">
        <p style="font-size:18px; margin-left:50px"> Instructions: </p>
        <p id="instruct"> Instruction info </p>
      </div>
    </div> <br>
  </div>


  <div class="container">
    <br> <br>
    <h5 id="rev" class="blue-text"> Reviews </h5>
    <div class="row">
      <div class="col s8">
        <ul class="collection">
          <li class="collection-item avatar">
            <img src="assets/images/profilepic.png" alt="" class="circle">
            <span class="title"> <a href="#"> Title </a></span>
            <p id="r-cmt"> User comment... </p>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div class="container">
    <p> Add a comment to review this recipe </p>
    <div class="row">
      <form class="col s6">
        <div class="input-field ">
          <label for="cmt"> Review: <i class="material-icons left">mode_edit</i></label>
          <textarea id="cmt" class="materialize-textarea " placeholder="Comment"></textarea>
        </div>
        <div class="right-align">
          <button class="btn grey" type="button"> Review </button>
        </div>
      </form>
    </div>
  </div>



  <footer class="page-footer teal lighten-2">
    <div class="container">
      <div class="row">
        <div class="col l6 s12">
          <h5 class="white-text">Company Bio</h5>
          <p class="grey-text text-lighten-4"> Information about the company, how it started, purpose and goal</p>


        </div>
        <div class="col l3 s12">
          <h5 class="white-text">About Us</h5>
          <ul>
            <li><a class="white-text" href="#!">Link 1</a></li>
            <li><a class="white-text" href="#!">Link 2</a></li>
          </ul>
        </div>
        <div class="col l3 s12">
          <h5 class="white-text">Privacy Settings</h5>
          <ul>
            <li><a class="white-text" href="#!">Link 1</a></li>
            <li><a class="white-text" href="#!">Link 2</a></li>
          </ul>
        </div>
      </div>
    </div>
    <div class="footer-copyright">
      <div class="container">
        Made by <a class="teal-text text-lighten-3">Osaze Edo-Osagie</a>
      </div>
    </div>
  </footer>


  <!--  Scripts-->
  <script src="https://code.jquery.com/jquery-2.1.1.min.js"></script>
  <script src="assets/js/materialize.js"></script>
  <script src="assets/js/init.js"></script>

</body>

</html>
