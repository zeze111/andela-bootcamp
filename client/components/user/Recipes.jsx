import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import swal from 'sweetalert';

/** Recipes class contains the list item for a single recipe
 *
 * @class Recipes
 *
 * @extends {React.Component}
 */
class Recipes extends Component {
  /**
   * @memberof Home
   *
   * @return {void}
   */
  clickEvent = () => {
    swal({
      text: 'Are you sure you want to delete this recipe',
      icon: 'warning',
      buttons: true,
      dangerMode: true,
    })
      .then((willDelete) => {
        if (willDelete) {
          this.props.deleteRecipe(this.props.recipe.id);
          swal('Poof! Your recipe has been deleted!', {
            icon: 'success',
          });
        }
      });
  }

  /** html component to render
   *
   * @memberof Home
   *
   * @return {void}
   */
  render() {
    return (
      <li className="collection-item">
        <Link
          to={`/recipe/${this.props.recipe.id}`}
          href={`/recipe/${this.props.recipe.id}`}
          className="text-color break caps"
        > {this.props.recipe.name}
        </Link>
        <div className="secondary-content div-pointer2">
          <i
            role="button"
            tabIndex="0"
            onKeyDown={this.handleKeyDown}
            onClick={this.clickEvent}
            className="material-icons icon-color"
          >delete
          </i>
        </div>
        <Link
          to={`/update-recipe/${this.props.recipe.id}`}
          href={`/update-recipe/${this.props.recipe.id}`}
          className="secondary-content"
        >
          <i className="material-icons icon-color" >mode_edit</i>
        </Link>
      </li>
    );
  }
}

Recipes.propTypes = {
  deleteRecipe: PropTypes.func.isRequired,
  recipe: PropTypes.objectOf(PropTypes.any),
};

Recipes.defaultProps = {
  recipe: {}
};

export default Recipes;
