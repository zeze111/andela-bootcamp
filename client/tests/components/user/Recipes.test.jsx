import React from 'react';

import Recipes from '../../../components/user/Recipes';
import store from '../../../store';

const props = {
  deleteRecipe: jest.fn(() => Promise.resolve()),
  recipe: {
    id: 1,
    name: 'blah',
    User: {
      id: 20,
      firstName: 'Jane'
    }
  },
};

const setup = () => {
  const wrapper = shallow(<Recipes {...props} store={store} />);
  return { wrapper }
}

describe('clickEvent', () => {
  it('should be called on delete a review', () => {
    const { wrapper } = setup();

    wrapper.instance().clickEvent();
  });
});

describe('Test for Recipes', () => {
  it('should render component', () => {

    const wrapper = setup();
    expect(wrapper).toMatchSnapshot()
  });
});
