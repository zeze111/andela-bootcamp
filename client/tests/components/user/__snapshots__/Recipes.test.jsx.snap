// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Recipes should render component 1`] = `
Object {
  "wrapper": <li
    className="collection-item"
  >
    <Link
      className="text-color break caps"
      href="/recipe/1"
      replace={false}
      to="/recipe/1"
    >
       
      blah
    </Link>
    <div
      className="secondary-content div-pointer2"
      data-place="bottom"
      data-tip="delete"
    >
      <i
        className="material-icons icon-color"
        onClick={[Function]}
        role="button"
        tabIndex="0"
      >
        delete
      </i>
    </div>
    <Link
      className="secondary-content"
      data-place="bottom"
      data-tip="edit"
      href="/update-recipe/1"
      replace={false}
      to="/update-recipe/1"
    >
      <i
        className="material-icons icon-color"
      >
        mode_edit
      </i>
    </Link>
    <ReactTooltip
      insecure={true}
      resizeHide={true}
      wrapper="div"
    />
  </li>,
}
`;
